src/main/java/com/github/mc1arke/sonarqube/plugin/scanner/CommunityProjectPullRequestsLoader.java

```diff
@@ -72,8 +73,10 @@ public CommunityProjectPullRequestsLoader(ScannerWsClient scannerWsClient) {
             } catch (ParseException e) {
                 LOGGER.warn("Could not parse date from Pull Requests API response. Will use '0' date", e);
             }
+            final JsonElement baseJson = jsonObject.get("base");
+            final String base = baseJson != null ? baseJson.getAsString() : null;
             return new PullRequestInfo(jsonObject.get("key").getAsString(), jsonObject.get("branch").getAsString(),
-                                       jsonObject.get("base").getAsString(), parsedDate);
+                    base, parsedDate);
```

Comment:
This could probably have been more succinctly written as `Optional.ofNullable(jsonObject.get("base")).orElse(null)` which would prevent the need for variable assignment and additional null checks.

Reply:
the getAsString transformation in your code example is missing, which makes it much longer and unreadable if not in an extra var

Reply:
I've got no issue with it being in an extra var, it was the null checks that seemed unnecessary. Your change looks good to me. I presume you've verified this now works for you locally?

Reply:
it works in production on our server

Reply:
and the nullcheck is necessary to prevent the NPEs. Otherwise you cannot call getAsString. Optional.orElse does nothing differently